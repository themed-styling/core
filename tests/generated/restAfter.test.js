import 'jest-styled-components'

import { restAfter } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(restAfter, 'restAfter')
