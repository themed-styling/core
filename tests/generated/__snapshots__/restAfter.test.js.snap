// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restAfter renders 0 1`] = `
Array [
  .c0 {
  rest-after: 0;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 0;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 0;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders function 1`] = `
Array [
  .c0 {
  rest-after: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: has;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: has;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders invalid theme path 1`] = `
Array [
  .c0 {
  rest-after: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders negative number 1`] = `
Array [
  .c0 {
  rest-after: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: -42365;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: -42365;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders object [breakpoints + standard] 1`] = `
Array [
  .c0 {
  rest-after: July;
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  rest-after: July;
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  rest-after: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    rest-after: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: June;
  }
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    rest-after: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    rest-after: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    rest-after: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    rest-after: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    rest-after: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    rest-after: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    rest-after: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders object [mediaQueries + standart] 1`] = `
Array [
  .c0 {
  rest-after: July;
}

@media May {
  .c0 {
    rest-after: August;
  }
}

@media April {
  .c0 {
    rest-after: September;
  }
}

@media January {
  .c0 {
    rest-after: November;
  }
}

@media April {
  .c0 {
    rest-after: August;
  }
}

@media June {
  .c0 {
    rest-after: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  rest-after: July;
}

@media May {
  .c0 {
    rest-after: August;
  }
}

@media April {
  .c0 {
    rest-after: September;
  }
}

@media January {
  .c0 {
    rest-after: November;
  }
}

@media April {
  .c0 {
    rest-after: August;
  }
}

@media June {
  .c0 {
    rest-after: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  rest-after: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    rest-after: November;
  }
}

@media April {
  .c0 {
    rest-after: January;
  }
}

@media January {
  .c0 {
    rest-after: December;
  }
}

@media April {
  .c0 {
    rest-after: April;
  }
}

@media June {
  .c0 {
    rest-after: May;
  }
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    rest-after: November;
  }
}

@media April {
  .c0 {
    rest-after: January;
  }
}

@media January {
  .c0 {
    rest-after: December;
  }
}

@media April {
  .c0 {
    rest-after: April;
  }
}

@media June {
  .c0 {
    rest-after: May;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders object [mediaQueries] 1`] = `
Array [
  @media May {
  .c0 {
    rest-after: July;
  }
}

@media April {
  .c0 {
    rest-after: August;
  }
}

@media January {
  .c0 {
    rest-after: September;
  }
}

@media April {
  .c0 {
    rest-after: November;
  }
}

@media June {
  .c0 {
    rest-after: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    rest-after: July;
  }
}

@media April {
  .c0 {
    rest-after: August;
  }
}

@media January {
  .c0 {
    rest-after: September;
  }
}

@media April {
  .c0 {
    rest-after: November;
  }
}

@media June {
  .c0 {
    rest-after: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    rest-after: January;
  }
}

@media April {
  .c0 {
    rest-after: December;
  }
}

@media January {
  .c0 {
    rest-after: April;
  }
}

@media April {
  .c0 {
    rest-after: May;
  }
}

@media June {
  .c0 {
    rest-after: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    rest-after: January;
  }
}

@media April {
  .c0 {
    rest-after: December;
  }
}

@media January {
  .c0 {
    rest-after: April;
  }
}

@media April {
  .c0 {
    rest-after: May;
  }
}

@media June {
  .c0 {
    rest-after: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders postitive number 1`] = `
Array [
  .c0 {
  rest-after: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 42365;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: 42365;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders string 1`] = `
Array [
  .c0 {
  rest-after: July;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: July;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: June;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: June;
}

<div
    className="c0"
  />,
]
`;

exports[`restAfter renders undefined 1`] = `
Array [
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
]
`;

exports[`restAfter renders valid theme path 1`] = `
Array [
  .c0 {
  rest-after: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  rest-after: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
